src/utils/hooks.js

```diff
@@ -19,3 +20,27 @@ export function useStateWithStorage(storageKey, initialValue) {
 
 	return [value, setValue];
 }
+
+export function purchaseUrgency(dateNextPurchased, lastPurchased) {
+	const today = new Date();
+	const nextPurchaseDaysBetween = getDaysBetweenDates(
+		today,
+		dateNextPurchased.toDate(),
+	);
+	const lastPurchaseDaysBetween = getDaysBetweenDates(
+		lastPurchased.toDate(),
+		today,
+	);
+	if (lastPurchaseDaysBetween > 60) {
+		return 'Inactive';
+	}
+	if (nextPurchaseDaysBetween < 0) {
+		return 'Overdue';
+	} else if (nextPurchaseDaysBetween <= 7) {
+		return 'Soon';
+	} else if (nextPurchaseDaysBetween <= 30) {
+		return 'Kind Of Soon';
+	} else {
+		return 'Not Soon';
+	}
```

Comment:
question: I realized the days that we assigned in `manageList.jsx` are different from here. I figured it out when I tried to add the `Not Soon` item. Is it expected to be like this?

![CleanShot 2024-03-15 at 09 43 54](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/22878284/80abc2a7-ebd9-4032-aa78-ae1202841e95)

Reply:
Hmm, good point on the Not Soon item! I do think we need to actually change the Not Soon operator from <=30 days to < 30, but this is what our issue said we needed to do to label the items:

<img width="570" alt="image" src="https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/137969744/2fd1be03-732b-4ae1-b2aa-d009d4fdca8f">

It's a little bit different from what the user originally inputs, as anything between 7 and 30 days is considered not soon. (At least by the requirements of our issue). I will change the operator for the 30 day one though!